### PR TITLE
Export a data map using fides resources

### LINKED ISSUE #324: Export Data Map as csv
### Is your feature request related to a specific problem?

As part of the journey to exporting a ROPA from `fidesctl`, the export of a data map is an important step. The data map should be a single csv that contains information about the organization, classifications of data processed, and where the data is transferred to (if applicable). 

Once the features are implemented in #321, #322, & #323 fides should be able to export a functional data map.

### Describe the solution you'd like

A single command to export a data map, something along the lines of `fidesctl export datamap` which in turn outputs a single csv made up of the following information:

**Controller organizational contact information (example provided recreated below, open to suggestions / alternatives)**

| Controller | Details | DPO | Details | Representative | Details |
| --- | --- | --- | --- | --- | --- |
| Name | _example name_ | Name | _example name_ | Name | _example name_ |
| Address | _example address_ | Address | _example address_ | Address | _example address_ |
| Telephone | _example telephone_ | Telephone | _example telephone_ | Telephone | _example telephone_ |
| Email | _example email_ | Email | _example email_ | Email | _example email_ |

**Data map details as rows (data points to be listed in sequence due to width)**

_Further refinement required still to ensure all requirements for the data map are met_

* Business Function? (Not certain this will be captured yet)
* Purpose of processing (`data_use`)
* Joint controller (if applicable)
* Categories of individuals (`data_subject`)
* Categories of personal data (`data_categories`)
* Categories of recipients (if applicable)
* Third countries transferred to (if applicable)
* Safeguards for transfer to third country (if applicable)
* Retention schedule (if possible)
* Link to security policy
* Source of personal data

### Describe alternatives you've considered, if any

Due to formatting done in the example, other formats could potentially be considered (example is `.xlsx` for instance) - csv seems to be a hard requirement so far however

### Additional context

The completion of this issue will close out the walk state of the data map project. The three prerequisite issues will enable this to be a reality, some flexibility in breakdown and inclusion of items may be expected here as we move into the model updates.

### PR DESCRIPTION
**Pull request by SteveDMurphy**
Closes #324 

### Code Changes

* [x] Use the provided example to generate output for the `Organization` resource as a `csv`
* [x] Create a new CLI command to export a data map
* [x] Use the provided example to build the "walk-state" output of a data map formatted for Excel
* [x] Ensure all components of the datamap can be separately exported as `csv`

### Steps to Confirm

* [x] Data output aligns with vision laid out in example data map / RoPA
* [x] Change to upsert functionality aligns with expectations of existing cli commands

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* [x] Documentation Updated

### Description Of Changes

This PR is focused on exporting the combined resources as a data map.

To test out the functionality, you should be able to do the following:

```
make cli
fidesctl apply demo_resources
fidesctl export datamap demo_resources
```

Additionally, the following flags can be applied to not use the excel format:
`fidesctl export datamap demo_resources --csv`
`fidesctl export datamap demo_resources --dry`

As well as having a separate command to output the organization resource separately as `csv` (all in support of those that do not want the formatted `.xlsx`)
`fidesctl export organization demo_resources`

The exported datamap is currently sent to the same directory housing the resource manifests where it can be viewed. This will likely need to be changed (or have a flag added) when ready to have this available for download from a fides UI.

Due to the output formatting, an Excel output was chosen (`.xlsx`) and uses a static copy of the form we were provided.

Attributes that are either yet to be modeled (or still to be exported properly as of writing) are either left blank or marked as `N/A`

Do we want to add another guide to the docs for exporting resources? I could go either way here, I think so but I also think it might be a bit light at this point and could possibly wait for the next release on the run state 🤷🏽

### CODE REVIEW

**Comment by SteveDMurphy** [comment]
@iamkelllly I am still going to be cleaning up a few things and sorting through docs but would love to get your thoughts on how this is working so far 🙌🏽

File: fidesctl/src/fidesapi/database/crud.py
Comment (SteveDMurphy) [inline fidesctl/src/fidesapi/database/crud.py:149]: I'm very much on the fence with this change here, I think it sort of defeats the purpose of using upsert except in the case of the default taxonomy. As far as I can tell however, it is the only use we have of upsert as of today... Would love an outside opinion here

Comment (ThomasLaPiana) [reply fidesctl/src/fidesapi/database/crud.py:149]: if someone runs `fidesctl db reset` we still want to give them a clean slate, but otherwise i don't think the upsert should be overwriting anything the user has done, and therefore I think this change makes sense.

so to be clear on what I think is okay here:
* `fidesctl db reset` wipes the whole database and adds all new stuff as per usual
* spinning up the webserver should _not_ overwrite what's already in the database (by using the on_conflict_do_nothing)

does this make sense/sound right?

Comment (SteveDMurphy) [reply fidesctl/src/fidesapi/database/crud.py:149]: Thanks for the second opinion! Yes that makes sense to me and I confirmed it works as you stated 🙌🏽

**Comment by SteveDMurphy** [comment]
> @iamkelllly I am still going to be cleaning up a few things and sorting through docs but would love to get your thoughts on how this is working so far 🙌🏽

Feedback captured via slack and addressed in here (fix issues with xlsx template, third country support). Joint controller only remaining gap, thinking about deferring until run state until we get more information

File: fidesctl/tests/core/test_dataset.py
Comment (SteveDMurphy) [reply fidesctl/tests/core/test_dataset.py]: Confirmed this is a MacOS specific issue, issue updated

File: docs/fides/docs/language/resources/organization.md
Comment (ThomasLaPiana) [inline docs/fides/docs/language/resources/organization.md:5]:
```suggestion
The Organization resource will also contain vital information with regards to compliance reporting in the case of a data map or RoPA (Record of Processing Activities).
```

File: fidesctl/src/fidesapi/database/crud.py
Comment (ThomasLaPiana) [reply fidesctl/src/fidesapi/database/crud.py:149]: perfect, thank you!

File: fidesctl/src/fidesctl/core/export.py
Comment (ThomasLaPiana) [inline fidesctl/src/fidesctl/core/export.py:494]: comment cleanup :D

**Review by ThomasLaPiana — APPROVED** [review]
LGTM! Had a suggestion as a small nit and would also recommend breaking out the `export.py` file, running up on 500 lines starts to get a bit unwieldy but these are non-blocking nits :)

File: fidesctl/src/fidesctl/core/export.py
Comment (SteveDMurphy) [reply fidesctl/src/fidesctl/core/export.py]: 🤦🏽 thanks!

**Comment by SteveDMurphy** [comment]
> LGTM! Had a suggestion as a small nit and would also recommend breaking out the `export.py` file, running up on 500 lines starts to get a bit unwieldy but these are non-blocking nits :)

Thanks for the review @ThomasLaPiana ! all updates made, split out the extras into an `export_helpers.py` (unsure on name but easy to change if we want)

**Comment by ThomasLaPiana** [comment]
@SteveDMurphy LGTM, merge when ready!

**Pull request merged**
